python/ray/util/state/state_manager.py

```diff
@@ -335,9 +335,38 @@ async def get_all_placement_group_info(
 
     @handle_grpc_network_errors
     async def get_all_node_info(
-        self, timeout: int = None
+        self,
+        timeout: int = None,
+        limit: int = None,
+        filters: Optional[List[Tuple[str, PredicateType, SupportedFilterType]]] = None,
     ) -> Optional[GetAllNodeInfoReply]:
-        request = GetAllNodeInfoRequest()
+        if not limit:
+            limit = RAY_MAX_LIMIT_FROM_DATA_SOURCE
+
+        if filters is None:
+            filters = []
+
+        req_filters = GetAllNodeInfoRequest.Filters()
+        for filter in filters:
+            logger.error(f"Filter: {filter}")
+            key, predicate, value = filter
+            if predicate != "=":
+                # We only support EQUAL predicate for source side filtering.
+                continue
```

Comment:
We should not silently drop the filters -- we'd fail instead if not supported filter is provided

Reply:
those filters are python-filtered. See https://github.com/ray-project/ray/blob/f13f0dee60a34e9da24bb83cba8f82b3848284e4/python/ray/dashboard/state_aggregator.py#L335

Reply:
Understood. We still shouldn't swallow any input coming from the API -- if we can't handle it we should fail loud and clear rather than silently swallowing it

Reply:
we have loud complaints in python side: https://github.com/ray-project/ray/blob/f13f0dee60a34e9da24bb83cba8f82b3848284e4/python/ray/dashboard/state_aggregator.py#L180

Reply:
It doesn't matter what's upstream -- this API is a standalone API and should not be ignoring the inputs -- if we can't handle the input make sure we fail the request.

Instead add filtering upstream in Python to avoid passing filters not supported in GCS

Reply:
hi this is not feasible in this PR alone - we have a versatile filtering API that is essentially *untyped* and each proto field can subject to filtering. I fear it's not best practice to add each and every field in each proto to the GCS side. https://github.com/ray-project/ray/blob/e6874e47ba9a6c05580b9863a93f79dd3c91520f/python/ray/util/state/common.py#L138

Reply:
to the end it's exposed to the ray CLI so users can do any filterings they want. So we can only:

1. breaking change, disallow some filters
2. do all filterings on GCS, 1 for each proto field in all of Node, Worker, Actor, etc
3. do "vital" filters that we (Dashboard, Ray) cares about in GCS, and others in Python

so this PR goes through (3).

Reply:
Spoke offline:

 - My comment was misplaced, this is the right way to handle it